app/models/events/search.rb

```diff
@@ -17,7 +17,7 @@ class Search
 
     validates :type, presence: false, inclusion: { in: :available_event_type_ids, allow_nil: true }
     validates :distance, inclusion: { in: :available_distance_keys }, allow_nil: true
-    validates :postcode, presence: true, postcode: { allow_blank: true }, if: :distance
+    validates :postcode, presence: true, postcode: { allow_blank: true, outward_only_postcode: true }, if: :distance
```

Comment:
The naming of this made me do a double take initially, as I thought it meant it would _only_ allow an outward postcode but in fact it allows an outward or full postcode. I would maybe rename it to something like `accept_partial_postcode`, but I don't feel strongly about it (I don't think that suggestion is actually all that much better, but there's maybe something clearer we can use).